### PR TITLE
feat: [PL-39761]: added AIDA setting and validator

### PR DESCRIPTION
**Pull request by bhavya181**
## Describe your changes
added AIDA setting and validator

Tested following locally -
1. Enabling AIDA on account scope, without signing EULA - throw error
2. Enabling AIDA on account scope, with EULA signed - enabled successfully
3. Enabling AIDA on project scope, when AIDA is disabled on Account scope - throws error
4. Enabling AIDA on project scope, when AIDA is enabled on Account scope - enabled successfully
5. Disable AIDA on both scopes

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51214)

### CODE REVIEW

File: 120-ng-manager/modules/ng-settings/src/main/java/io/harness/ngsettings/services/impl/validators/EnableAidaValidator.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      validateIfSettingIsEnabledOnAccountScope(accountIdentifier, newSetting.getIdentifier());
+    } else {
+      validateIfEulaIsSigned(accountIdentifier, newSetting.getIdentifier());
+    }
+  }
+
+  private void validateIfSettingIsEnabledOnAccountScope(String accountIdentifier, String settingIdentifier) {
+    SettingValueResponseDTO responseDTO = settingsService.get(settingIdentifier, accountIdentifier, null, null);
+    if (!Boolean.parseBoolean(responseDTO.getValue())) {
+      throw new InvalidRequestException(
```
Comment (gn-harness) [inline 120-ng-manager/modules/ng-settings/src/main/java/io/harness/ngsettings/services/impl/validators/EnableAidaValidator.java:38]: Do we need it to be enabled at the account scope? Can't we just enable at the project level?

Comment (gn-harness) [reply 120-ng-manager/modules/ng-settings/src/main/java/io/harness/ngsettings/services/impl/validators/EnableAidaValidator.java:38]: Is this what we mandate for other settings also?

File: 120-ng-manager/modules/ng-settings/src/main/java/io/harness/ngsettings/services/impl/validators/EnableAidaValidator.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    if (isNotEmpty(newSetting.getProjectIdentifier())) {
+      validateIfSettingIsEnabledOnAccountScope(accountIdentifier, newSetting.getIdentifier());
+    } else {
+      validateIfEulaIsSigned(accountIdentifier, newSetting.getIdentifier());
+    }
+  }
+
+  private void validateIfSettingIsEnabledOnAccountScope(String accountIdentifier, String settingIdentifier) {
+    SettingValueResponseDTO responseDTO = settingsService.get(settingIdentifier, accountIdentifier, null, null);
+    if (!Boolean.parseBoolean(responseDTO.getValue())) {
```
Comment (gn-harness) [inline 120-ng-manager/modules/ng-settings/src/main/java/io/harness/ngsettings/services/impl/validators/EnableAidaValidator.java:37]: What is setting is already disabled on account and we are turning project level setting off? Is that even a valid case?

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51214)* status: 7 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/modules/ng-settings/src/main/java/io/harness/ngsettings/services/impl/validators/EnableAidaValidator.java` line 37 at r1](https://reviewable.io/reviews/harness/harness-core/51214#gh-1281213811:-Naq0Z51BEYElhSfyfGA:b-ud9pqe) ([raw file](https://github.com/harness/harness-core/blob/b48f2c3a5c3cf4f2641ada0f21b9c10bcce6b56b/120-ng-manager/modules/ng-settings/src/main/java/io/harness/ngsettings/services/impl/validators/EnableAidaValidator.java#L37)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

What is setting is already disabled on account and we are turning project level setting off? Is that even a valid case?
</blockquote></details>

That is not possible, if it is disabled on Account, it will be disabled on project as well. 
___
*[`120-ng-manager/modules/ng-settings/src/main/java/io/harness/ngsettings/services/impl/validators/EnableAidaValidator.java` line 38 at r1](https://reviewable.io/reviews/harness/harness-core/51214#gh-1281213245:-Naq06ihBgSfmGYI5KhH:b-yfgue) ([raw file](https://github.com/harness/harness-core/blob/b48f2c3a5c3cf4f2641ada0f21b9c10bcce6b56b/120-ng-manager/modules/ng-settings/src/main/java/io/harness/ngsettings/services/impl/validators/EnableAidaValidator.java#L38)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

Is this what we mandate for other settings also?
</blockquote></details>

No, we do not have this behaviour for other settings, but this behaviour is specifically asked by PM that we cannot enable AIDA on project if its not enabled on Account

**Review by gn-harness — DISMISSED** [review]

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51214)* status: 7 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/modules/ng-settings/src/main/resources/io/harness/ngsettings/settings.yml` line 443 at r1](https://reviewable.io/reviews/harness/harness-core/51214#-NatjjjQ3aFKaBPLhclg:-NatjjjR91bKIb2Gm2VM:b-5ebuoy) ([raw file](https://github.com/harness/harness-core/blob/b48f2c3a5c3cf4f2641ada0f21b9c10bcce6b56b/120-ng-manager/modules/ng-settings/src/main/resources/io/harness/ngsettings/settings.yml#L443)):*
> ```YAML
>     allowedScopes:
>       - ACCOUNT
>       - PROJECT
> ```

Is it intentional to skip org scope here?

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51214)* status: 7 file(s) to review. 1 change requests. 3. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/modules/ng-settings/src/main/resources/io/harness/ngsettings/settings.yml` line 443 at r1](https://reviewable.io/reviews/harness/harness-core/51214#-NatjjjQ3aFKaBPLhclg:-NavaBnlEOeH5YByhaRk:bab3ypj) ([raw file](https://github.com/harness/harness-core/blob/b48f2c3a5c3cf4f2641ada0f21b9c10bcce6b56b/120-ng-manager/modules/ng-settings/src/main/resources/io/harness/ngsettings/settings.yml#L443)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

Is it intentional to skip org scope here?
</blockquote></details>

Yes, setting will be allowed only on Account and project scope

**Review by piyush-harness — APPROVED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51214)* status: Some of the checks are too old. 1 change requests. 3. unresolved discussions.

**Comment by bhavya181** [comment]
trigger feature-build

**Comment by bhavya181** [comment]
trigger platform_svc

**Comment by bhavya181** [comment]
trigger uts

**Review by nishantsaini20 — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51214)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained.

**Comment by bhavya181** [comment]
trigger ut9

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51214)* status: 1 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Pull request merged**